### PR TITLE
Fix Proxmox builder unhandled buildvar type

### LINKED ISSUE #10139: Proxmox builder unhandled buildvar type: map[interface {}]interface {}
#### Overview of the Issue

Proxmox builder returns **unhandled buildvar type: map[interface {}]interface {}** when using with HCL

### Packer version

1.6.4
Works well on 1.6.0

### Simplified Packer Buildfile

[Source](https://github.com/blz-ea/proxmox-packer/blob/master/ubuntu-server-18-04/sources.pkr.hcl)

### Operating system and Environment details

Ubuntu 18.04

### Log Fragments and crash.log files

```
2020/10/20 09:09:59 [INFO] Packer version: 1.6.4 [go1.15.2 linux amd64]
2020/10/20 09:09:59 Checking 'PACKER_CONFIG' for a config file path
2020/10/20 09:09:59 'PACKER_CONFIG' not set; checking the default config file path
2020/10/20 09:09:59 Attempting to open config file: //home/debian/.packerconfig
2020/10/20 09:09:59 [WARN] Config file doesn't exist: /home/debian/.packerconfig
2020/10/20 09:09:59 Setting cache directory: /home/debian/packer_cache
1.6.4
2020/10/20 09:09:59 [INFO] (telemetry) Finalizing.
2020/10/20 09:09:59 waiting for all plugin processes to complete...
2020/10/20 09:09:59 [INFO] Packer version: 1.6.4 [go1.15.2 linux amd64]
2020/10/20 09:09:59 Checking 'PACKER_CONFIG' for a config file path
2020/10/20 09:09:59 'PACKER_CONFIG' not set; checking the default config file path
2020/10/20 09:09:59 Attempting to open config file: /home/debian/.packerconfig
2020/10/20 09:09:59 [WARN] Config file doesn't exist: /home/debian/.packerconfig
2020/10/20 09:09:59 Setting cache directory: /home/debian/packer_cache
2020/10/20 09:09:59 Creating plugin client for path: /home/debian/packer
2020/10/20 09:09:59 Starting plugin: /home/debian/packer []string{"/home/debian/packer", "plugin", "packer-builder-proxmox"}
2020/10/20 09:09:59 Waiting for RPC address for: /home/debian/packer
2020/10/20 09:09:59 packer-builder-proxmox plugin: [INFO] Packer version: 1.6.4 [go1.15.2 linux amd64]
2020/10/20 09:09:59 packer-builder-proxmox plugin: Checking 'PACKER_CONFIG' for a config file path
2020/10/20 09:09:59 packer-builder-proxmox plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/10/20 09:09:59 packer-builder-proxmox plugin: Attempting to open config file: /home/debian/.packerconfig
2020/10/20 09:09:59 packer-builder-proxmox plugin: [WARN] Config file doesn't exist: /home/debian/.packerconfig
2020/10/20 09:09:59 packer-builder-proxmox plugin: Setting cache directory: /home/debian/packer_cache
2020/10/20 09:09:59 packer-builder-proxmox plugin: args: []string{"packer-builder-proxmox"}
2020/10/20 09:09:59 packer-builder-proxmox plugin: Plugin address: unix /tmp/packer-plugin254088721
2020/10/20 09:09:59 packer-builder-proxmox plugin: Waiting for connection...
2020/10/20 09:09:59 Received unix RPC address for /home/debian/packer: addr is /tmp/packer-plugin254088721
2020/10/20 09:09:59 packer-builder-proxmox plugin: Serving a plugin connection...
2020/10/20 09:09:59 packer-builder-proxmox plugin: CPU type not set, using default 'kvm64'
2020/10/20 09:09:59 packer-builder-proxmox plugin: Disk 0 cache mode not set, using default 'none'
2020/10/20 09:10:00 Creating plugin client for path: /home/debian/packer
2020/10/20 09:10:00 Starting plugin: /home/debian/packer []string{"/home/debian/packer", "plugin", "packer-provisioner-shell"}
2020/10/20 09:10:00 Waiting for RPC address for: /home/debian/packer
2020/10/20 09:10:00 packer-provisioner-shell plugin: [INFO] Packer version: 1.6.4 [go1.15.2 linux amd64]
2020/10/20 09:10:00 packer-provisioner-shell plugin: Checking 'PACKER_CONFIG' for a config file path
2020/10/20 09:10:00 packer-provisioner-shell plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/10/20 09:10:00 packer-provisioner-shell plugin: Attempting to open config file: /home/debian/.packerconfig
2020/10/20 09:10:00 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/debian/.packerconfig
2020/10/20 09:10:00 packer-provisioner-shell plugin: Setting cache directory: /home/debian/packer_cache
2020/10/20 09:10:00 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2020/10/20 09:10:00 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin273311822
2020/10/20 09:10:00 packer-provisioner-shell plugin: Waiting for connection...
2020/10/20 09:10:00 Received unix RPC address for /home/debian/packer: addr is /tmp/packer-plugin273311822
2020/10/20 09:10:00 packer-provisioner-shell plugin: Serving a plugin connection...
2020/10/20 09:10:00 Build debug mode: true
2020/10/20 09:10:00 Force build: false
2020/10/20 09:10:00 On error: 
2020/10/20 09:10:00 Debug enabled, so waiting for build to finish: proxmox.template
2020/10/20 09:10:00 Starting build run: proxmox.template
2020/10/20 09:10:00 Running builder: 
2020/10/20 09:10:00 [INFO] (telemetry) Starting builder 
Debug mode enabled. Builds will not be parallelized.
proxmox.template: output will be in this color.

2020/10/20 09:10:00 packer-builder-proxmox plugin: No URLs were provided to Step Download. Continuing...
==> proxmox.template: Pausing after run of step 'StepDownload'. Press enter to continue. 
==> proxmox.template: Pausing after run of step 'stepUploadISO'. Press enter to continue. 
==> proxmox.template: Pausing after run of step 'stepUploadAdditionalISOs'. Press enter to continue. 
==> proxmox.template: Creating VM
==> proxmox.template: Starting VM
==> proxmox.template: Pausing after run of step 'stepStartVM'. Press enter to continue. 
2020/10/20 09:10:04 packer-builder-proxmox plugin: Found available port: 8137 on IP: 0.0.0.0
==> proxmox.template: Starting HTTP server on port 8137
==> proxmox.template: Pausing after run of step 'StepHTTPServer'. Press enter to continue. 
==> proxmox.template: Waiting 10s for boot
==> proxmox.template: Typing the boot command
2020/10/20 09:10:15 packer-builder-proxmox plugin: [INFO] Waiting 1s
2020/10/20 09:10:16 packer-builder-proxmox plugin: [INFO] Waiting 1s
2020/10/20 09:10:17 packer-builder-proxmox plugin: [INFO] Waiting 1s
==> proxmox.template: Pausing after run of step 'stepTypeBootCommand'. Press enter to continue. 
2020/10/20 09:10:28 packer-builder-proxmox plugin: [DEBUG] Unable to get address during connection step: 500 QEMU guest agent is not running
2020/10/20 09:10:28 packer-builder-proxmox plugin: [INFO] Waiting for SSH, up to timeout: 30m0s
==> proxmox.template: Waiting for SSH to become available...
2020/10/20 09:10:31 packer-builder-proxmox plugin: [DEBUG] Error getting SSH address: 500 QEMU guest agent is not running
...
2020/10/20 09:14:42 packer-builder-proxmox plugin: [DEBUG] Error getting SSH address: 500 QEMU guest agent is not running
2020/10/20 09:14:47 packer-builder-proxmox plugin: [INFO] Attempting SSH connection to 192.168.88.122:22...
2020/10/20 09:14:47 packer-builder-proxmox plugin: [DEBUG] reconnecting to TCP connection for SSH
2020/10/20 09:14:47 packer-builder-proxmox plugin: [DEBUG] handshaking with SSH
2020/10/20 09:14:47 packer-builder-proxmox plugin: [DEBUG] handshake complete!
2020/10/20 09:14:47 packer-builder-proxmox plugin: [DEBUG] Opening new ssh session
2020/10/20 09:14:48 packer-builder-proxmox plugin: [INFO] agent forwarding enabled
==> proxmox.template: Connected to SSH!
==> proxmox.template: Pausing after run of step 'StepConnect'. Press enter to continue. 
2020/10/20 09:15:40 packer-builder-proxmox plugin: Running the provision hook
2020/10/20 09:15:40 [INFO] (telemetry) Starting provisioner shell
==> proxmox.template: Pausing 20s before the next provisioner...
2020/10/20 09:16:00 [INFO] (telemetry) ending shell
==> proxmox.template: Provisioning step had errors: Running the cleanup provisioner, if present...
==> proxmox.template: Pausing before cleanup of step 'StepConnect'. Press enter to continue. 
==> proxmox.template: Pausing before cleanup of step 'stepTypeBootCommand'. Press enter to continue. 
==> proxmox.template: Pausing before cleanup of step 'StepHTTPServer'. Press enter to continue. 
==> proxmox.template: Pausing before cleanup of step 'stepStartVM'. Press enter to continue. 
==> proxmox.template: Stopping VM
==> proxmox.template: Deleting VM
==> proxmox.template: Pausing before cleanup of step 'stepUploadAdditionalISOs'. Press enter to continue. 
==> proxmox.template: Pausing before cleanup of step 'stepUploadISO'. Press enter to continue. 
==> proxmox.template: Pausing before cleanup of step 'StepDownload'. Press enter to continue. 
2020/10/20 09:16:11 [INFO] (telemetry) ending 
2020/10/20 09:16:11 Waiting on builds to complete...
==> Wait completed after 6 minutes 11 seconds
2020/10/20 09:16:11 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/10/20 09:16:11 machine readable: proxmox.template,error []string{"unhandled buildvar type: map[interface {}]interface {}"}
==> Builds finished but no artifacts were created.
2020/10/20 09:16:11 [INFO] (telemetry) Finalizing.
Build 'proxmox.template' errored after 6 minutes 11 seconds: unhandled buildvar type: map[interface {}]interface {}

==> Wait completed after 6 minutes 11 seconds

==> Some builds didn't complete successfully and had errors:
--> proxmox.template: unhandled buildvar type: map[interface {}]interface {}

==> Builds finished but no artifacts were created.
2020/10/20 09:16:11 waiting for all plugin processes to complete...
2020/10/20 09:16:11 packer: plugin process exited
2020/10/20 09:16:11 packer: plugin process exited
```

### PR DESCRIPTION
**Pull request by blz-ea**
Closes #10139

### CODE REVIEW

File: builder/proxmox/step_start_vm.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func (s *stepStartVM) Run(ctx context.Context, state multistep.StateBag) multist
 	state.Put("vmRef", vmRef)
 	// instance_id is the generic term used so that users can have access to the
 	// instance id inside of the provisioners, used in step_provision.
-	state.Put("instance_id", vmRef)
+	state.Put("instance_id", vmRef.VmId())
```
Comment (carlpett) [inline builder/proxmox/step_start_vm.go:92]: For the future us who wonder about this, let's add a small mental note :) 
```suggestion
	// Note that this is just the VMID, we do not keep the node, pool and other
	// info available in the vmref type.
	state.Put("instance_id", vmRef.VmId())
```

**Review by carlpett — COMMENTED** [review]
Thanks! I think we can just add a small comment, suggested some wording.

**Review by carlpett — APPROVED** [review]
Great, thanks a lot! Just the CLA thing remains, otherwise ready to merge :cake:

**Comment by SwampDragons** [comment]
yep, this looks good. I'll merge once the CLA is signed. Looks like this needs a rebase, too, now that we've merged that other Proxmox PR.

File: builder/proxmox/step_start_vm.go (hunk 1 above)
Comment (azr) [reply builder/proxmox/step_start_vm.go:94]: ah that was it, really cool fix !

**Review by azr — APPROVED** [review]
SuperCool Fix ! Thanks !

**Pull request merged**